cuda/cuda_nemolite2d.cu

```diff
@@ -319,6 +359,171 @@ cuda_initialise_grid_()
   cudaDeviceSynchronize();
 }
 
+void
+cuda_setup_model_params_(int jpi,
+                         int jpj,
+                         wp_t dx,
+                         wp_t dy,
+                         wp_t dep_const,
+                         int nit000,
+                         int nitend,
+                         int irecord,
+                         wp_t rdt,
+                         wp_t cbfr,
+                         wp_t visc)
+{
+  printf("[CUDA](Host) Initialising model params.\n");
+
+  model_params = {
+    .jpi = jpi,
+    .jpj = jpj,
+    .dx = dx,
+    .dy = dy,
+    .dep_const = dep_const,
+    .nit000 = nit000,
+    .nitend = nitend,
+    .irecord = irecord,
+    .rdt = rdt,
+    .cbfr = cbfr,
+    .visc = visc,
+  };
+}
+
+void
+cuda_continuity_()
+{}
+
+void
+cuda_momentum_()
+{
+  // TODO:
+}
+
+void
+cuda_boundary_conditions_(wp_t rtime)
+{
+  const int jpi = model_params.jpi;
+  const int jpj = model_params.jpj;
+
+  k_boundary_conditions<<<jpi + 1, jpj + 1>>>(rtime,
+
+                                              *simulation_vars.sshn_u,
+                                              *simulation_vars.sshn_v,
+
+                                              *simulation_vars.ssha,
+
+                                              *simulation_vars.ua,
+                                              *simulation_vars.va,
+
+                                              *simulation_vars.ua_buffer,
+                                              *simulation_vars.va_buffer,
+
+                                              *grid_constants.hu,
+                                              *grid_constants.hv,
+
+                                              *grid_constants.pt,
+
+                                              jpi,
+                                              jpj);
+
+  cudaDeviceSynchronize();
+
+  // Now swap the double buffered arrays.
+  wp_t* const ua_buffered_data =
+    simulation_vars.ua_buffer->get_device_data_ptr();
+  wp_t* const va_buffered_data =
+    simulation_vars.va_buffer->get_device_data_ptr();
+
+  simulation_vars.ua_buffer->set_device_data_ptr(
+    simulation_vars.ua->get_device_data_ptr());
+  simulation_vars.va_buffer->set_device_data_ptr(
+    simulation_vars.va->get_device_data_ptr());
+
+  simulation_vars.ua->set_device_data_ptr(ua_buffered_data);
+  simulation_vars.va->set_device_data_ptr(va_buffered_data);
+}
+
+void
+cuda_next_()
+{
+  const int jpi = model_params.jpi;
+  const int jpj = model_params.jpj;
+
+  k_next<<<jpi + 1, jpj + 1>>>(*simulation_vars.sshn,
```

Comment:
This could generate an invalid grid-block configuration. Also only a 1D kernel launch.